mlflow/server/js/src/common/utils/AuthComponent.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Utils from '../../common/utils/Utils';
+import { Redirect } from 'react-router';
+import PropTypes from 'prop-types';
+
+export class AuthComponentImpl extends Component {
+  constructor(props) {
+    super(props);
+    const urlState = Utils.getSearchParamsFromUrl(props.location.search);
+    this.state = {
+      persistedState: {
+        code: urlState.code === undefined ? '' : urlState.code,
+      },
+    };
+  }
+
+  static propTypes = {
+    location: PropTypes.object,
+  };
+
+  componentDidMount = () => {
+    const req = new XMLHttpRequest();
+    req.open('GET', '/token?code=' + this.state.persistedState.code, false);
+    req.send();
+    if (req.status === 200) {
+      const token = req.getResponseHeader(req.getResponseHeader('Access-Control-Expose-Headers'));
```

Comment:
I don t understand well this line, but the header in wich the token is stored is: X-JWT-Token

Reply:
But this header is stored in 'Access-Control-Expose-Headers' no ?

Reply:
Yes, [This header Access-Control-Expose-Headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers) is a bit special. You will not able to fetch the value of a "custom" header in the front if its name is not in "Access-Control-Expose-Headers". We may add other header in the future, and so this one (a list of header  name) can change. It is much safer here (and easier to read), to "hardcode" the header name

Reply:
Done